sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -281,27 +306,109 @@ export class ECertGenerationService {
    * Calculate the effective value for every award. The result of this calculation
    * will be the value used to generate the e-Cert.
    * @param disbursements all disbursements that are part of one e-Cert.
+   * @param entityManager used to execute the commands in the same transaction.
    */
-  private calculateEffectiveValue(disbursements: ECertDisbursementSchedule[]) {
+  private async calculateEffectiveValue(
+    disbursements: ECertDisbursementSchedule[],
+    entityManager: EntityManager,
+  ): Promise<void> {
     for (const disbursement of disbursements) {
       for (const disbursementValue of disbursement.disbursementValues) {
-        if (this.shouldStopFunding(disbursement, disbursementValue)) {
+        if (this.shouldStopFullTimeBCFunding(disbursement, disbursementValue)) {
```

Comment:
I believe the method here must be more Generic in name(as it was before) and action, not just to stop full time and only BC funding.